src/realm/sync/client.cpp

```diff
@@ -1865,6 +1980,7 @@ ClientImpl::Connection::Connection(ClientImpl& client, connection_ident_type ide
     , m_ssl_verify_callback{std::move(ssl_verify_callback)}               // DEPRECATED
     , m_proxy_config{std::move(proxy_config)}                             // DEPRECATED
     , m_reconnect_info{reconnect_info}
+    , m_session_history{20}
```

Comment:
This is creating a vector which contains 20 zeros, which doesn't seem like a sensible thing to be doing?

Reply:
I was thinking this would set the initial capacity to 20 so the majority of sessions wouldn't need to allocate additional memory, but, you're right, this is adding 20 default constructed elements to the vector. I should be calling `reserve()` instead.